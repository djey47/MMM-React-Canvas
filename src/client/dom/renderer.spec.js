import {
  renderWrapper,
  renderMainComponent,
} from './renderer.js';

describe('renderWrapper function', () => {
  it('should return correct HTML', () => {
    // given-when
    const actual = renderWrapper('IDWrapper');
    // then
    expect(actual).toMatchSnapshot();
  });
});

describe('renderMainComponent function', () => {
  it('should render correctly', () => {
    // given
    const wrapperId = 'IDWrapper';
    document.write(`<div id='${wrapperId}' />`);

    // when
    const rendered = renderMainComponent(wrapperId);

    // then
    expect(rendered).not.toBeUndefined();
  });
});
