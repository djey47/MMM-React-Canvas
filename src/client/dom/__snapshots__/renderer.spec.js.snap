// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderWrapper function should return correct HTML 1`] = `
<div
  class="IDWrapper"
  id="IDWrapper"
/>
`;
